codec-http2/src/main/java/io/netty/handler/codec/http2/Http2StreamChannelOption.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2025 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.http2;
+
+import io.netty.channel.ChannelOption;
+
+/**
+ * {@link ChannelOption}s that are specific to {@link Http2StreamChannel}s.
+ *
+ * @param <T>   the type of the value which is valid for the {@link ChannelOption}
+ */
+public final class Http2StreamChannelOption<T> extends ChannelOption<T> {
+    private Http2StreamChannelOption(String name) {
+        super(name);
+    }
+
+    /**
+     * When set to {@code true} {@link Http2WindowUpdateFrame}s will be automatically be generated and written for
+     * {@link Http2StreamChannel}s as soon as frames are passed to the user via
+     * {@link io.netty.channel.ChannelPipeline#fireChannelRead(Object)}. If the user wants more control on when a
+     * window update is send its possible to set it to {@code false}. In this case the user is responsible to
+     * generate the correct {@link Http2WindowUpdateFrame}s and eventually write these to the channel.
+     */
+    public static final ChannelOption<Boolean> AUTO_WRITE_WINDOW_UPDATE_FRAME =
+            valueOf("AUTO_WRITE_WINDOW_UPDATE_FRAME");
```

Comment:
Did you intend to make this a `Http2StreamChannelOption`? afaict, the `valueOf(..)` makes a standard ChannelOption, which is fine, but I'm not sure why `Http2StreamChannelOption` needs to extend `ChannelOption<T>` for anything other than looks since there will never be any instances.

Reply:
It's just the pattern we follow for others as well so I thought we should do here as well.